sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v2/script.sql

```diff
@@ -0,0 +1,415 @@
+      QUALIFY
+        rnk = 1
+    ),
+    --get all the details for that session from the day before the session date and the session date itself
+    event_aggregates AS (
+      SELECT
+        a.user_pseudo_id AS ga_client_id,
+        CAST(e.value.int_value AS string) AS ga_session_id,
+        COUNTIF(event_name = 'page_view') AS pageviews,
+        MIN(event_timestamp) AS min_event_timestamp,
+        MAX(event_timestamp) AS max_event_timestamp,
+        CAST(
+          MAX(CASE WHEN event_name = 'product_download' THEN 1 ELSE 0 END) AS boolean
+        ) AS had_download_event
+      FROM
+        `moz-fx-data-marketing-prod.analytics_313696158.events_*` a
+      JOIN
+        UNNEST(event_params) e
+      WHERE
+        _table_suffix
+        BETWEEN FORMAT_DATE('%Y%m%d', DATE_SUB(@submission_date, INTERVAL 1 DAY))
+        AND FORMAT_DATE('%Y%m%d', @submission_date)
+        AND e.key = 'ga_session_id'
+        AND e.value.int_value IS NOT NULL
+      GROUP BY
+        a.user_pseudo_id,
+        CAST(e.value.int_value AS string)
+    ),
+    stub_session_ids_staging AS (
+      SELECT
+        user_pseudo_id AS ga_client_id,
+        event_timestamp,
+        CAST(
+          (
+            SELECT
+              `value`
+            FROM
+              UNNEST(event_params)
+            WHERE
+              key = 'ga_session_id'
+            LIMIT
+              1
+          ).int_value AS string
+        ) AS ga_session_id,
+        CAST(e.value.int_value AS string) AS stub_session_id
+      FROM
+        `moz-fx-data-marketing-prod.analytics_313696158.events_*`
+      JOIN
+        UNNEST(event_params) e
+      WHERE
+        _TABLE_SUFFIX
+        BETWEEN FORMAT_DATE('%Y%m%d', DATE_SUB(@submission_date, INTERVAL 1 DAY))
+        AND FORMAT_DATE('%Y%m%d', @submission_date)
+        AND event_name = 'stub_session_set'
+        AND e.key = 'id'
+        AND e.value.int_value IS NOT NULL
+    ),
+    last_stub_session_id AS (
+      SELECT
+        ga_client_id,
+        event_timestamp,
+        ga_session_id,
+        stub_session_id,
+        ROW_NUMBER() OVER (
+          PARTITION BY
+            ga_client_id,
+            ga_session_id
+          ORDER BY
+            event_timestamp DESC
+        ) AS stub_session_rnk
+      FROM
+        stub_session_ids_staging
+      QUALIFY
+        stub_session_rnk = 1
+    ),
+    all_stub_session_ids AS (
+      SELECT
+        ga_client_id,
+        ga_session_id,
+        ARRAY_AGG(stub_session_id) AS all_reported_stub_session_ids
+      FROM
+        stub_session_ids_staging
+      GROUP BY
+        ga_client_id,
+        ga_session_id
+    ),
+    landing_page_by_session_staging AS (
+      SELECT
+        user_pseudo_id AS ga_client_id,
+        CAST(
+          (
+            SELECT
+              `value`
+            FROM
+              UNNEST(event_params)
+            WHERE
+              key = 'ga_session_id'
+            LIMIT
+              1
```

Comment:
That function would work for these as well: `mozfun.map.get_first(event_params, "ga_session_id")`